lib/hsa/mcwamp_hsa.cpp

```diff
@@ -271,6 +271,22 @@ static unsigned extractBits(unsigned v, unsigned pos, unsigned w)
 };
 
 
+
+
+namespace hc {
+
+// printf buffer size (in number of packets, see hc_printf.hpp)
+constexpr unsigned int default_printf_buffer_size = 2048;
+
+// address of the global printf buffer in host coherent system memory
+PrintfPacket* printf_buffer = nullptr;
```

Comment:
@scchan would it be possible to make it a member variable of `HSAContext`, as its life-cycle is tied to it?

Reply:
I'd prefer to leave it here.  One of the reasons is that the device function hc::printf has a reference to this global object and if I put this inside HSAContext then we will have to include the HSAContext definition when we compile device code, which I'd try to avoid.